kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/FilterBuilderImpl.java

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.kubernetes.client.dsl.base;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesResourceList;
+import io.fabric8.kubernetes.api.model.LabelSelector;
+import io.fabric8.kubernetes.api.model.LabelSelectorRequirement;
+import io.fabric8.kubernetes.api.model.ObjectReference;
+import io.fabric8.kubernetes.client.dsl.FilterBuilder;
+import io.fabric8.kubernetes.client.dsl.FilterWatchListDeletable;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.client.utils.Utils;
+
+import java.lang.reflect.Array;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+final class FilterBuilderImpl<T extends HasMetadata, L extends KubernetesResourceList<T>, R extends Resource<T>> implements FilterBuilder<FilterWatchListDeletable<T, L>> {
+  
+  private static final String INVOLVED_OBJECT_NAME = "involvedObject.name";
+  private static final String INVOLVED_OBJECT_NAMESPACE = "involvedObject.namespace";
+  private static final String INVOLVED_OBJECT_KIND = "involvedObject.kind";
+  private static final String INVOLVED_OBJECT_UID = "involvedObject.uid";
+  private static final String INVOLVED_OBJECT_RESOURCE_VERSION = "involvedObject.resourceVersion";
+  private static final String INVOLVED_OBJECT_API_VERSION = "involvedObject.apiVersion";
+  private static final String INVOLVED_OBJECT_FIELD_PATH = "involvedObject.fieldPath";
+  
+  private final BaseOperation<T, L, R> baseOperation;
+  private OperationContext context;
+
+  /**
+   * @param baseOperation
```

Comment:
nit, either complete this javadoc or remove it.

Reply:
No problem, I should probably turn that off in the ide...